storage/range_command.go

```diff
@@ -234,7 +263,7 @@ func (r *Range) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args *
 		}
 	} else {
 		// The transaction doesn't exist yet on disk; use the supplied version.
-		reply.Txn = gogoproto.Clone(args.Txn).(*proto.Transaction)
+		reply.Txn = args.Txn
```

Comment:
It's likely that this doesn't matter but it's implicit that methods shouldn't change the arguments unless they consciously chose to do that. Here we're setting two pointers equal and then changing away. If there's ever a bug here, it will be painful to find. Let's just keep the copy.